python/pyarrow/tests/test_array.py

```diff
@@ -1893,6 +1893,19 @@ def __arrow_array__(self, type=None):
     with pytest.raises(TypeError):
         pa.array(arr)
 
+    # ARROW-7066 - allow ChunkedArray output
+    class MyArray2:
+        def __init__(self, data):
+            self.data = data
+
+        def __arrow_array__(self, type=None):
+            return pa.chunked_array([self.data], type=type)
+
+    arr = MyArray2(np.array([1, 2, 3], dtype='int64'))
+    result = pa.array(arr)
+    expected = pa.array([1, 2, 3], type=pa.int64())
```

Comment:
Well, this also needs to be a chunked array here.
```suggestion
    expected = pa.chunked_array([[1, 2, 3]], type=pa.int64())
```

Reply:
Whoops, that was indeed what I wanted to test ;)